File: docs/snippet/code-signal/arcade/intro/land-of-logic/longest-word/longest-word.py
def longest_word(text):
    max_word, word = '', ''

    for char in text:
        if char.isalpha():
            word += char
        else:
            max_word = max(word, max_word, key=len)
            word = ''

    return max(word, max_word, key=len)


print(longest_word("abcd"))

File: content/snippet/code-signal/arcade/intro/land-of-logic/longest-word/longest-word.js
function longestWord(text) {
  let maxWord = '';
  let word = '';

  for (let char of text) {
    if (char.match(/[a-zA-Z]/)) {
      word += char;
    } else {
      maxWord = word.length > maxWord.length ? word : maxWord;
      word = '';
    }
  }

  return word.length > maxWord.length ? word : maxWord;
}

console.log(longestWord('Ready, steady, go!'));
